### DISCUSSION TITLE
Share the data and full route caches across multiple instances

### DISCUSSION THREAD

**Discussion by psebestyen** [vercel/next.js#55836] [Help] [answered]
### Summary

Caching and revalidation with revalidatePath and revalidateTag works fine if I serve my app from a single running instance. Most non-toy applications will need to run multiple instances for redundancy and for horizontal scaling. Some of these instances will need to run on multiple machines as well to guard against hardware failures. If the cache is not shared or synchronized across these instances, the app may serve stale data which may not be acceptable. Is there a general solution to this problem? Without a solution I have two options: 1) run my app in a single instance, 2) opt out of the nextjs data cache. Neither option is very appealing.  Any guidance would be greatly appreciated.

Thank you

**Comment by wwwdepot** [comment, marked as answer]
See https://nextjs.org/docs/app/api-reference/next-config-js/incrementalCacheHandlerPath

**Reply by psebestyen** [reply]
So with this API I can implement my own centralized caching. That's great. Do you know of any existing implementation? I was not able to find any.

**Reply by better-salmon** [reply]
@psebestyen hello! Check out this implementation of the custom cache handler.

https://caching-tools.github.io/next-shared-cache/
